pycaret/utils/_dependencies.py

```diff
@@ -1,15 +1,87 @@
 # Adapted from
 # https://github.com/alan-turing-institute/sktime/blob/v0.11.0/sktime/utils/validation/_dependencies.py
 
+import collections
 import sys
+from distutils.version import LooseVersion
 from importlib import import_module
-from typing import Optional
+from typing import Dict, Optional, Union
 
-from pycaret.internal.logging import get_logger
-from pycaret.utils._show_versions import _get_module_version
+from importlib_metadata import distributions
+
+from pycaret.internal.logging import get_logger, redirect_output
 
 logger = get_logger()
 
+INSTALLED_PACKAGES = None
+
+
+def _try_import_and_get_module_version(
+    modname: str,
+) -> Optional[Union[LooseVersion, bool]]:
+    """Returns False if module is not installed, None if version is not available"""
+    try:
+        if modname in sys.modules:
+            mod = sys.modules[modname]
+        else:
+            if logger:
+                with redirect_output(logger):
+                    mod = import_module(modname)
+            else:
+                mod = import_module(modname)
+        try:
+            ver = mod.__version__
+        except AttributeError:
+            # Version could not be obtained
+            ver = None
+    except ImportError:
+        ver = False
+    if ver:
+        ver = LooseVersion(ver)
+    return ver
+
+
+def get_installed_packages() -> Dict[str, Optional[LooseVersion]]:
+    global INSTALLED_PACKAGES
+    if not INSTALLED_PACKAGES:
+        # Get all installed modules and their versions without
+        # needing to import them.
+        pkg_to_dist = collections.defaultdict(list)
+        for dist in distributions():
+            for pkg in (dist.read_text("top_level.txt") or "").split():
+                try:
+                    ver = LooseVersion(dist.metadata["Version"])
+                except Exception:
+                    ver = None
+                pkg_to_dist[pkg] = ver
+        INSTALLED_PACKAGES = dict(pkg_to_dist)
+    return INSTALLED_PACKAGES
+
+
+def _get_module_version(modname: str) -> Optional[Union[LooseVersion, bool]]:
+    """Will cache the version in INSTALLED_PACKAGES"""
+    installed_packages = get_installed_packages()
+    if modname not in installed_packages:
+        # Fallback. This should never happen.
+        installed_packages[modname] = _try_import_and_get_module_version(modname)
+    return installed_packages[modname]
+
+
+def get_module_version(modname: str) -> Optional[LooseVersion]:
+    """Raises a ValueError if module is not installed"""
+    version = _get_module_version(modname)
+    if version is False:
```

Comment:
I don't see how this can ever be False.  ` _get_module_version` returns dict entries with either a version or None

Reply:
`_try_import_and_get_module_version` will return False